rfcs/20221006-frontend_event_tracking.md

```diff
@@ -0,0 +1,469 @@
+   *   - On what page do users typically find a result?
+   *   - How often and how many pages of results do users load?
+   *   - Can we experiment with the types of results / result rankings
+   *     on certain pages, pages that users don't usually choose a result
+   *     from anyway?
+   */
+  LOAD_MORE_RESULTS: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+    /** The current page of results the user is on. */
+    resultPage: number;
+  };
+  /**
+   * Description: When a user opens the external sources popover.
+   * Questions:
+   *   - How often do users use this feature?
+   *   - Under what conditions to users use this feature? No results?
+   *     Many results, but none they actually select?
+   */
+  VIEW_EXTERNAL_SOURCES: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+    /** The current page of results the user is on. */
+    resultPage: number;
+  };
+  /**
+   * Description: When the user chooses an external source from the dropdown of external sources
+   * Questions:
+   *   - Which external sources are most popular? This could drive inclusion in Openverse.
+   *   - Are certain media types more popular externally?
+   */
+  SELECT_EXTERNAL_SOURCE: {
+    /** The name of the external source */
+    name: string;
+    /** The full URL of the source */
+    url: string;
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term */
+    query: string;
+  };
+  /**
+   * Description: The user opens the menu which lists pages.
+   * Questions:
+   *   - How often is this menu used?
+   *   - Is this menu visible enough?
+   */
+  OPEN_PAGES_MENU: {};
+  /**
+   * Description: The user clicks to a link outside of Openverse.
+   * Questions:
+   *   - What types of external content do users seek?
+   *   - Are there external resources we should make more visible?
+   *   - Is there content we might want to add to Openverse directly?
+   */
+  EXTERNAL_LINK_CLICK: {
+    /** The url of the external link */
+    url: string;
+    /**
+     * The name of the Vue component used to switch content types.
+     * @todo: We need to determine the best way of creating a string
+     * representation of a vue component, which allows us to know
+     * which ui element was clicked. Perhaps we could use unique data-test-ids
+     * or another unique identifier for this?
+     */
+    component: string;
+  };
+  /**
+   * Description: Whenever the user selects a result from the search results page.
+   * Questions:
+   *   - Which results are most popular for given searches?
+   *   - How often do searches lead to clicking a result?
+   *   - Are there popular searches that do not result in result selection?
+   */
+  SELECT_SEARCH_RESULT: {
+    /** If the result is a related result, provide the ID of the 'original' result */
+    related: string | null;
```

Comment:
It does! I've fixed it to include that. `related` will refer to when a piece of media is selected from a single result  view:

![CleanShot 2023-01-12 at 14 55 50](https://user-images.githubusercontent.com/6351754/212168560-c83c8a3d-82fe-4f3c-b1bb-bae6dbf6d32a.png)

Reply:
A potentially clearer name: `relatedTo` rather than `related` which does sound to me more like it'd be whatever the related image was, not the original.

Reply:
^1000 times what Sara said, please.